rascal-lsp/src/main/rascal/util/LanguageServer.rsc

```diff
@@ -268,6 +272,10 @@ data LanguageService
     | references    (set[loc] (Focus _focus) referencesService)
     | implementation(set[loc] (Focus _focus) implementationService)
     | codeAction    (list[CodeAction] (Focus _focus) codeActionService)
+    | prepareRename (loc (Focus _focus) prepareRenameService)
+    | rename        (tuple[list[DocumentEdit], set[Message]] (Focus _focus, str newName) renameService)
```

Comment:
Publishing the diagnostic messages resulting from the rename overlaps with `util::IDEServices::registerDiagnostics`. Can we reduce this overlap? We still want to abort a rename when an `error` was registered, even when edits were registered before the error.

*(also applies to `willRenameFiles`)*

Reply:
that discussion on errors in a operation is broader than this api. normally we use exceptions for true errors.

Reply:
So, I am not sure if I understad - would it be fine to return a `set[Message]` here?

Reply:
I think we want to associate the reasons for failure to locations (for example, we cannot rename because that would cause shadowing at `x`). That's where the `Message` comes in handy.

Reply:
I think that there is an orthogonal discussion about how we report back errors from any language contribution. That is not specific to `rename` only.

Like what if I want to report an error when someone requests a `definition` or a `codeAction`?

Reply:
I see. I'll leave this how it is for now.